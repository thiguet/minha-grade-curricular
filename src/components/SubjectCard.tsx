import React from 'react';

export interface SubjectCardProps {
    name: string;
    acronym: string;
    hours: number;
    onClick: () => {};
}

export default function Card({ name, hours, acronym, onClick }: SubjectCardProps) {
    return (
        <div>
            <div className="close-btn" onClick={onClick}>
                <img src="" />
            </div>
            <p className="subject-name">{name}</p>
            <p className="acronym">{acronym}</p>
            <p className="hours">{hours}</p>
        </div>
    );
}
