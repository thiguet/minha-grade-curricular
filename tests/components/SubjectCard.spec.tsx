import React from 'react';
import renderer from 'react-test-renderer';
import SubjectCard, { SubjectCardProps } from '../../src/components/SubjectCard';
import { mount } from 'enzyme';
import { getFakeSubjectCardProps } from '../fake/ComponentProps';

describe('Subject Card', () => {
    let props: SubjectCardProps;

    const build = () => {
        const wrapper = mount(<SubjectCard {...props} />);
        const snapRenderer = renderer.create(<SubjectCard {...props} />);
        return {
            wrapper,
            snapRenderer,
            subjectName: () => wrapper.find('.subject-name'),
            hours: () => wrapper.find('.hours'),
            acronym: () => wrapper.find('.acronym'),
            closeBtn: () => wrapper.find('.close-btn'),
        };
    };

    beforeEach(() => {
        props = getFakeSubjectCardProps();
    });
    it('must render the component as expected', () => {
        props = {
            name: 'fixed name',
            hours: 20,
            acronym: 'FN',
            onClick: jest.fn(),
        };
        const { snapRenderer } = build();
        expect(snapRenderer.toJSON()).toMatchSnapshot();
    });

    it('must render main components', () => {
        const { subjectName, hours, acronym } = build();

        expect(subjectName()).toBeTruthy();
        expect(hours()).toBeTruthy();
        expect(acronym()).toBeTruthy();
    });

    it('must pass props to main main components', () => {
        const { subjectName, hours, acronym } = build();

        expect(subjectName().text()).toBe(props.name);
        expect(hours().text()).toBe(`${props.hours}`);
        expect(acronym().text()).toBe(`${props.acronym}`);
    });

    it('must simulate a click from the close button component', () => {
        const { closeBtn } = build();

        closeBtn().simulate('click');

        expect(props.onClick).toBeCalled();
    });
});
