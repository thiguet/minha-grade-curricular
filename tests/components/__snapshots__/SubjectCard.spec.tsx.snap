// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subject Card must render the component as expected 1`] = `
<div>
  <div
    className="close-btn"
    onClick={[MockFunction]}
  >
    <img
      src=""
    />
  </div>
  <p
    className="subject-name"
  >
    fixed name
  </p>
  <p
    className="acronym"
  >
    FN
  </p>
  <p
    className="hours"
  >
    20
  </p>
</div>
`;
